src/coreclr/jit/optimizer.cpp

```diff
@@ -2874,38 +2874,49 @@ bool Compiler::optIsLoopEntry(BasicBlock* block) const
     return false;
 }
 
-// Canonicalize the loop nest rooted at parent loop 'loopInd'.
-// Returns 'true' if the flow graph is modified.
+//-----------------------------------------------------------------------------
+// optCanonicalizeLoop: Canonicalize a loop nest
+//
+// Arguments:
+//   loopInd - index of outermost loop in the nest
+//
+// Returns:
+//   true if the flow graph was modified
+//
+// Notes:
+//   For loopInd and all contained loops, ensures each loop top's back edges
+//   do not come from nested loops.
+//
+//   Will split top blocks and redirect edges if needed.
+//
 bool Compiler::optCanonicalizeLoopNest(unsigned char loopInd)
 {
-    bool modified = false;
-
-    // Is the top of the current loop in any nested loop?
-    if (optLoopTable[loopInd].lpTop->bbNatLoopNum != loopInd)
-    {
-        if (optCanonicalizeLoop(loopInd))
-        {
-            modified = true;
-        }
-    }
+    bool modified = optCanonicalizeLoop(loopInd);
```

Comment:
I didn't understand why the check of `if (optLoopTable[loopInd].lpTop->bbNatLoopNum != loopInd)` was removed although the comment says `"For loopInd and all contained loops, ensures each loop top's back edges do not come from nested loops."`

Reply:
We were checking this redundantly; now we just check it down in `optCanonicalizeLoop`.

I may move that check back up to the caller with my next round of revisions, but the point is we don't need to check it twice.

Reply:
ah..see it now.